pkg/scheduler/cache/cache.go

```diff
@@ -51,7 +51,7 @@ type schedulerCache struct {
 	period time.Duration
 
 	// This mutex guards all fields within this cache struct.
-	mu sync.Mutex
+	mu sync.RWMutex
```

Comment:
The goal of this change is to improve `cache.IsUpToDate`'s cost.

Reply:
Did we forget to approve/merge the PR where someone else made this change? /=

Reply:
Nope, that previous PR only improved `ecache`, not `cache` :D

Reply:
So, the way I see it, we are making 2 changes as part of this PR,

- Changing scheduler cache's lock from Mutex to RWMutex
- Changing ecache to be 2 level cache.

So, the flame graphs attached, could be result of both the changes? Am I missing something?

Reply:
The performance improvement is brought by 2 level cache. Cache refactoring is just a small bonus :D